packages/extension-api/src/extension-api.d.ts

```diff
@@ -1834,6 +1858,70 @@ declare module '@podman-desktop/api' {
     export const clipboard: Clipboard;
   }
 
+  /**
+   * Represents a dictionary object with string keys and values of type T.
+   */
+  export interface Dict<T> {
+    /**
+     * Index signature for accessing values by keys.
+     */
+    [key: string]: T | undefined;
+  }
+
+  /**
+   * Options for running a command.
+   */
+  export interface RunOptions {
+    /**
+     * Environment variables to set for the command.
+     */
+    env?: Dict<string>;
+  }
+
+  /**
+   * Represents the result of running a command.
+   */
+  export interface RunResult {
+    /**
+     * The exit code of the command.
+     */
+    exitCode: number;
+
+    /**
+     * The standard output (stdout) content of the command.
+     */
+    stdOut: string;
+
+    /**
+     * The standard error (stderr) content of the command.
+     */
+    stdErr: string;
+  }
+
+  /**
+   * Namespace for environment-related utilities.
+   */
+  export namespace exec {
+    /**
+     * Run provided command, returning an object that contains the exit code, stdout, and stderr content.
+     * Does not reject the promise if the exit code is not 0.
+     * @param command The command to execute.
+     * @param args The command arguments.
+     * @param options Options, such as environment variables.
+     * @returns A promise that resolves to a RunResult object.
+     */
+    export function runCliCommand(command: string, args: string[], options?: RunOptions): Promise<RunResult>;
+
+    /**
+     * Execute provided command, returning the stdout content. Rejects the promise if the exit code is not 0.
+     * @param command The command to execute.
+     * @param args The command arguments.
+     * @param options Options, such as environment variables.
+     * @returns A promise that resolves to the stdout content as a string.
+     */
+    export function execCliCommand(command: string, args: string[], options?: RunOptions): Promise<string>;
```

Comment:
probably avoid this API as return type is too limited, won't scale.

people can do `const {stdout} = await runCliCommand(...)`